vendor/views/vendor.py

```diff
@@ -267,11 +259,11 @@ def post(self, request, *args, **kwargs):
             return redirect('vendor:checkout-review')
 
 
-class ReviewCheckoutView(LoginRequiredMixin, TemplateView):
+class ReviewCheckoutView(LoginRequiredMixin, SetSiteToRequestMixin, TemplateView):
     template_name = 'vendor/checkout.html'
 
     def get(self, request, *args, **kwargs):
-        invoice = get_purchase_invoice(request.user)
+        invoice = get_purchase_invoice(request.user, request.site)
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
invoice = get_purchase_invoice(request.user, get_site_from_request(request))